### PR TITLE
[Enhancement] Support display datacache metircs by http service.

### PR DESCRIPTION
**Pull request by GavinMar**
Why I'm doing:
We need some more metrics to evaluate the datacache performance. But now it is difficult because we only expose the datacache memory tracker to trace the memory usage.

What I'm doing:
* Update the starcache version which returns more metrics from the api.
* Expose more detailed cache metrics by http service to help observe datacache states.

With this PR, we can get detailed datacache metrics by the following URL:
```
http://${BE_HOST}:${BE_WEBSERVER_PORT}/api/datacache/stat
```
The response looks like below:
```json
{
"status": "NORMAL",
"mem_quota_bytes": 5368709120,
"mem_used_bytes": 4237890114,
"disk_quota_bytes": 214748364800,
"disk_used_bytes": 0,
"mem_used_rate": 0.79,
"disk_used_rate": 0,
"disk_spaces": "/home/disk1/datacache:214748364800",
"meta_used_bytes": 3123454,
"hit_count": 114438,
"miss_count": 1126,
"hit_rate": 0.99,
"hit_bytes": 29809132182,
"miss_bytes": 268505666,
"hit_count_last_minute": 114061,
"miss_count_last_minute": 1126,
"hit_bytes_last_minute": 29713540072,
"miss_bytes_last_minute": 268505666,
"read_mem_bytes": 29809132182,
"read_disk_bytes": 0,
"write_bytes": 4237890114,
"write_success_count": 16268,
"write_fail_count": 20856,
"remove_bytes": 0,
"remove_success_count": 0,
"remove_fail_count": 0,
"current_reading_count": 2,
"current_writing_count": 0,
"current_removing_count": 0
}
```

## What type of PR is this:

- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: be/src/block_cache/block_cache.cpp [hunk 1]
```diff
@@ -168,8 +168,8 @@ void BlockCache::record_read_cache(size_t size, int64_t lateny_us) {
     _kv_cache->record_read_cache(size, lateny_us);
 }
 
-const DataCacheMetrics BlockCache::cache_metrics() const {
-    return _kv_cache->cache_metrics();
+const DataCacheMetrics BlockCache::cache_metrics(int level) const {
```
Comment (Smith-Cruise) [inline be/src/block_cache/block_cache.cpp:171]: can we use enum instead?

File: be/src/http/CMakeLists.txt [hunk 2]
```diff
@@ -53,6 +53,7 @@ add_library(Webserver STATIC
   action/update_config_action.cpp
   action/runtime_filter_cache_action.cpp
   action/query_cache_action.cpp
+  action/datacache_action.cpp
```
Comment (Smith-Cruise) [inline be/src/http/CMakeLists.txt:56]: datacache_metrics_action better?

File: be/src/block_cache/block_cache.cpp (hunk 1 above)
Comment (GavinMar) [reply be/src/block_cache/block_cache.cpp:171]: > can we use enum instead?

It is mainly to be consistent with the dependent library interface, and we can replace it after the library is updated.

**Review by Smith-Cruise — APPROVED** [review]

File: be/src/http/action/datacache_action.cpp [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+            if (!block_cache || !block_cache->is_initialized()) {
+                _handle_error(req, strings::Substitute("Cache system is not ready"));
+            } else {
+                _handle_stat(req, block_cache);
+            }
+        } else {
+            _handle_error(req, strings::Substitute("Not support GET method: '$0'", req->uri()));
+        }
+    } else {
+        _handle_error(req,
```
Comment (kevincai) [inline be/src/http/action/datacache_action.cpp:68]: usually if the method is not allowed, it should be a HTTP 405 error. not a HTTP 200 with response body `error: not supported ....`

**Review by kevincai — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Pull request merged**